src/learning-catalogue/model/factory/moduleFactory.ts

```diff
@@ -66,6 +68,16 @@ export class ModuleFactory {
 			module.events = (data.events || []).map(this.eventFactory.create)
 			module.productCode = data.productCode
 
+			module.events.sort((event1: Event, event2: Event) => {
+				if (!event1.dateRanges[0]) {
+					return 1
+				} else if (!event2.dateRanges[0]) {
+					return -1
+				} else {
+					return DateTime.sortDateRanges(event1.dateRanges[0], event2.dateRanges[0])
+				}
```

Comment:
a candidate for ternary operator perhaps?
```suggestion
				return !event1.dateRanges[0] ? 1 : !event2.dateRanges[0] ? -1 : DateTime.sortDateRanges(event1.dateRanges[0], event2.dateRanges[0])
```